src/commands/ext-dev-deprecate.ts

```diff
@@ -0,0 +1,77 @@
+import * as clc from "cli-color";
+
+import { Command } from "../command";
+import { promptOnce } from "../prompt";
+import { ensureExtensionsApiEnabled, logPrefix } from "../extensions/extensionsHelper";
+import { deprecateExtensionVersion, listExtensionVersions } from "../extensions/extensionsApi";
+import * as refs from "../extensions/refs";
+import { parseVersionPredicate, compareVersions } from "../extensions/versionHelper";
+import { requireAuth } from "../requireAuth";
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+
+/**
+ * Deprecate all extension versions that match the version predicate.
+ */
+export default new Command("ext:dev:deprecate <extensionRef> <versionPredicate>")
+  .description("deprecate extension versions that match the version predicate")
+  .option("-m, --message <deprecationMessage>", "deprecation message")
+  .option(
+    "-f, --force",
+    "override deprecation message for existing deprecated extension versions that match"
+  )
+  .before(requireAuth)
+  .before(ensureExtensionsApiEnabled)
+  .action(async (extensionRef: string, versionPredicate: string, options: any) => {
+    const { publisherId, extensionId, version } = refs.parse(extensionRef);
+    const baseExtensionRef = publisherId + "/" + extensionId;
+    if (version) {
+      throw new FirebaseError(
+        `The input extension reference must be of the format ${clc.bold(
+          "<publisherId>/<extensionId>"
+        )}. Version should be supplied in the version predicate argument.`
+      );
+    }
+    if (!publisherId || !extensionId) {
+      throw new FirebaseError(
+        `Error parsing publisher ID and extension ID from extension reference '${clc.bold(
+          extensionRef
+        )}'. Please use the format '${clc.bold("<publisherId>/<extensionId>")}'.`
+      );
+    }
+    const { comparator, targetSemVer } = parseVersionPredicate(versionPredicate);
+    const extensionVersions = await listExtensionVersions(baseExtensionRef);
```

Comment:
I'm ok with this approach - however, the backend does support > and < list filtering for ExtensionVersions. We use it for displaying releaseNotes in the CLI if you want an example

Reply:
Oh yeah, this is way better. Thanks for the suggestion!